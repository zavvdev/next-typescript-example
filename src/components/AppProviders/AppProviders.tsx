import React from 'react';
import { Provider } from 'react-redux';
import { AppProvidersProps } from 'src/components/AppProviders/AppProviders.types';

const AppProviders = (props: AppProvidersProps) => {
  const {
    children,
    store,
  } = props;

  return (
    <Provider store={store}>
      {children}
    </Provider>
  );
};

export default AppProviders;
