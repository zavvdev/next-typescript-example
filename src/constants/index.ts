export const APP_LANGUAGES = {
  en: 'en',
  ru: 'ru',
};
