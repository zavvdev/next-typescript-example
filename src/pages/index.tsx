import type { GetStaticProps, NextPage } from 'next';
import { serverSideTranslations } from 'next-i18next/serverSideTranslations';
import HomeView from 'src/views/Home/Home.view';
import I18N_NAMESPACES from 'src/constants/i18nNamespaces';

const Home: NextPage = () => {
  return (
    <HomeView />
  );
};

export const getStaticProps: GetStaticProps = async ({ locale }) => {
  return {
    props: {
      ...(await serverSideTranslations(
        locale as string,
        [I18N_NAMESPACES.home, I18N_NAMESPACES.common],
      )),
    },
  };
};

export default Home;
