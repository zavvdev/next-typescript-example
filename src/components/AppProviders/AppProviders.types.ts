import { ReactNode } from 'react';
import { configureStore } from 'src/store';

export interface AppProvidersProps {
  children: ReactNode;
  store: ReturnType<typeof configureStore>;
}
