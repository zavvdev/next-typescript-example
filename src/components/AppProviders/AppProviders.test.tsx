import { screen } from '@testing-library/react';
import { configureStore } from 'src/store';
import render from 'src/tests/utils/render';
import AppProviders from 'src/components/AppProviders/AppProviders';
import { AppProvidersProps } from 'src/components/AppProviders/AppProviders.types';

const childrenMock = 'children_mock';

const appProvidersProps: AppProvidersProps = {
  store: configureStore(),
  children: (
    <div>{childrenMock}</div>
  ),
};

describe('<AppProviders />', () => {
  test('should render AppProviders component (snapshot)', () => {
    const { asFragment } = render(
      <AppProviders {...appProvidersProps} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('should have a children', () => {
    render(<AppProviders {...appProvidersProps} />);
    expect(screen.getByText(childrenMock)).toBeInTheDocument();
  });
});
