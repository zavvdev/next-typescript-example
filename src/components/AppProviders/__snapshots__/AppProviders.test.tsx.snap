// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppProviders /> should render AppProviders component (snapshot) 1`] = `
<DocumentFragment>
  <div>
    children_mock
  </div>
</DocumentFragment>
`;
